sdk/src/transaction_context.rs

```diff
@@ -251,6 +304,40 @@ impl TransactionContext {
     pub fn get_instruction_trace(&self) -> &InstructionTrace {
         &self.instruction_trace
     }
+
+    /// Calculates the sum of all lamports within an instruction
+    pub fn instruction_accounts_lamport_sum(
+        &self,
+        instruction_accounts: &[InstructionAccount],
+    ) -> Result<u128, InstructionError> {
+        if !self.is_early_verification_of_account_modifications_enabled() {
+            return Ok(0);
+        }
+        let mut instruction_accounts_lamport_sum: u128 = 0;
+        for (instruction_account_index, instruction_account) in
+            instruction_accounts.iter().enumerate()
+        {
+            if instruction_account_index != instruction_account.index_in_callee {
+                continue; // Skip duplicate account
+            }
+            instruction_accounts_lamport_sum = (self
+                .get_account_at_index(instruction_account.index_in_transaction)?
+                .try_borrow()
+                .map_err(|_| InstructionError::AccountBorrowOutstanding)?
```

Comment:
the `AccountBorrowOutstanding` check starts to be ommitted for program_accounts it seems: https://github.com/solana-labs/solana/blob/3e9bd6170c9d78db850c9c7c7bcd832b4ce311bd/program-runtime/src/invoke_context.rs#L476-L481

Reply:
Good finding, seems like this was only enforced in `verify()` (at transaction instruction level), not in CPI (`verify_and_update()`). Nonetheless, I included it to happen in every instruction `pop()`.